knotx-core/knotx-api/src/main/java/com/cognifide/knotx/api/RepositoryResponse.java

```diff
@@ -3,92 +3,65 @@
  *
  * Copyright (C) 2016 Cognifide Limited
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
  *
- *      http://www.apache.org/licenses/LICENSE-2.0
+ * http://www.apache.org/licenses/LICENSE-2.0
  *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
  */
 package com.cognifide.knotx.api;
 
 import io.vertx.core.json.JsonObject;
 import io.vertx.rxjava.core.MultiMap;
 import rx.Observable;
 
-public class RepositoryResponse extends JsonObjectRequest {
+public abstract class RepositoryResponse extends JsonObjectRequest {
 
     private static final long serialVersionUID = -5892594714118271978L;
 
-    private boolean success;
+    protected String data;
 
-    private String reason;
+    protected int statusCode;
 
-    private String data;
+    protected MultiMap headers;
 
-    private int statusCode;
+    public static RepositoryResponse fromJson(JsonObject object) {
+        RepositoryResponse repositoryResponse;
 
-    private MultiMap headers;
+        MultiMap headers = fromJsonArray(object.getJsonArray("headers"));
+        Boolean shouldProcess = object.getBoolean("shouldProcess");
+        int statusCode = object.getInteger("statusCode");
+        String data = object.getString("data");
 
-    private RepositoryResponse() {
-        //Empty default constructor
-    }
-
-    public RepositoryResponse(JsonObject object) {
-        success = object.getBoolean("success");
-        headers = fromJsonArray(object.getJsonArray("headers"));
-        if (success) {
-            data = object.getString("data");
+        if (statusCode == 200) {
```

Comment:
This check does not match criteria:
`If statusCode=200 and no body - return as it is (empty body) keeping the headers and statusCode from repository`.
Currently variable `shouldProcess` is not used anywhere. Shouldn't be here:
```java
if (shouldProcess) {
    repositoryResponse = success(data, headers);
} else {
....
```
?